imap_processing/glows/packet_definitions/P_GLX_TMSCHIST.xml

```diff
@@ -0,0 +1,190 @@
+<?xml version='1.0' encoding='utf-8'?>
+<xtce:SpaceSystem xmlns:xtce="http://www.omg.org/space/xtce" name="P_GLX_TMSCHIST">
+	<xtce:Header date="2023-09" version="1.0" author="IMAP SDC" />
+	<xtce:TelemetryMetaData>
+		<xtce:ParameterTypeSet>
+			<xtce:IntegerParameterType name="UINT1" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="1" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT2" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="2" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT3" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="3" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT8" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="8" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT11" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="11" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT14" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="14" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT16" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="16" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT24" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="24" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="UINT32" signed="false">
+				<xtce:IntegerDataEncoding sizeInBits="32" encoding="unsigned" />
+			</xtce:IntegerParameterType>
+			<xtce:IntegerParameterType name="INT32" signed="true">
+				<xtce:IntegerDataEncoding sizeInBits="32" encoding="signed" />
+			</xtce:IntegerParameterType>
+			<xtce:BinaryParameterType name="BYTE28800">
+				<xtce:BinaryDataEncoding bitOrder="mostSignificantBitFirst">
+					<xtce:SizeInBits>
+						<xtce:FixedValue>28800</xtce:FixedValue>
```

Comment:
Good, point, as far as I am aware, it is a fixed value. I have gotten some test data and this successfully decoms it. It's a standard size for the number of histogram buckets, so I think it will stay the same, but I will double-check when I meet with them next.

Reply:
With Ultra there were some nuances that I didn't figure out until I was given the Flight Software Specifications Table. Even after that we had to walk through their L0 decom code in C for a couple of hours and only then it became clear what was going on. My suggested questions for them:

1. Should the decom data be exactly the same as the simulator data (I do think that it should be)
2. Should this be a variable length packet?
3. Do they take only the first x values from the returned array? Ultra had hard-coded [0:48] into their C code to select only the first 48 values, which I never would have known about if I hadn't been walked through what they were doing.

I might just be paranoid from my own experience so maybe its all exactly as it should be :) I just recommend that you double check.

Reply:
Definitely will! Thank you for those suggested questions, that's helpful for understanding your concerns and what I should look into.